src/main/java/org/prebid/server/bidder/compass/CompassBidder.java

```diff
@@ -0,0 +1,173 @@
+import java.util.stream.Collectors;
+
+public class CompassBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpCompass>> COMPASS_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public CompassBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> modifiedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpCompass extImpCompass = parseImpExt(imp);
+                modifiedImps.add(modifyImp(imp, extImpCompass));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+        final List<HttpRequest<BidRequest>> outgoingRequests = modifiedImps.stream()
+                .map(imp -> createSingleRequest(imp, request))
+                .collect(Collectors.toList());
+
+        return Result.of(outgoingRequests, errors);
+    }
+
+    private HttpRequest<BidRequest> createSingleRequest(Imp imp, BidRequest request) {
+        final BidRequest outgoingRequest = request.toBuilder().imp(Collections.singletonList(imp)).build();
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
+                .build();
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpCompass extImpCompass) {
+        try {
+            return imp.toBuilder()
+                    .ext(mapper.mapper().valueToTree(RequestBodyExt.of(getImpExtCompassWithType(extImpCompass))))
+                    .build();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private ImpExtCompass getImpExtCompassWithType(ExtImpCompass extImpCompass) {
+        ImpExtCompass impExtCompass = null;
+        if (!Objects.equals(extImpCompass.getEndpointId(), "")) {
+            impExtCompass = ImpExtCompass.builder()
+                    .type("network")
+                    .endpointId(extImpCompass.getEndpointId())
+                    .build();
+        } else if (!Objects.equals(extImpCompass.getPlacementId(), "")) {
+            impExtCompass = ImpExtCompass.builder()
+                    .type("publisher")
+                    .placementId(extImpCompass.getPlacementId())
+                    .build();
+        }
+        return impExtCompass;
+    }
+
```

Comment:
You can directly return `RequestBodyExt`.
Nitpick: rename method to `buildImpExt`.